modules/uhttp/client/client_filters.go

```diff
@@ -81,6 +82,20 @@ func tracingFilter(
 	return resp, err
 }
 
+func authenticationFilter(
+	ctx fx.Context, req *http.Request, next BasicClient,
+) (resp *http.Response, err error) {
+	authClient := uauth.Instance()
+	authctx, err := authClient.Authenticate(ctx)
+	if err != nil {
+		ctx.Logger().Error(uauth.ErrAuthentication, "error", err)
```

Comment:
sorry I meant to add this to uhttp/filters.go, we have access to the host there.